samtranslator/model/eventsources/scheduler.py

```diff
@@ -138,8 +138,17 @@ def _construct_scheduler_schedule_without_target(
         if self.State:
             scheduler_schedule.State = self.State
 
-        # Scheduler schedule's Name is a required property
-        scheduler_schedule.Name = self.Name or self.logical_id
+        if self.Name:
+            scheduler_schedule.Name = self.Name
+        elif self.Name is False:
+            # When `Name` is `False` explicitly,
+            # we set `AWS::Scheduler::Schedule`'s Name to None
+            # and `Name` is not specified, to keep backward compatibility,
+            # we use name generated by SAM instead of passing `None`:
+            # https://github.com/aws/serverless-application-model/issues/3109
```

Comment:
This comment is a bit confusing; the second part of the sentence (starting with "and \`Name\` is not specified")  refers to behavior in a different branch of the conditional. 
It might be clearer if you move the whole comment to right before the `if`, or move the second part of the sentence to the `else` block.

Reply:
good idea

Reply:
updated

Reply:
no longer applicable